src/cljstyle/format/core.clj

```diff
@@ -177,9 +177,13 @@
       shebang
       (update :formatted (partial str shebang))
 
-      (and (get-in rules-config [:eof-newline :enabled?])
-           (not (str/ends-with? (:formatted result) "\n")))
-      (update :formatted str "\n"))))
+      (get-in rules-config [:eof-newline :enabled?])
+      (cond->
+        (not (str/ends-with? (:formatted result) "\n"))
+        (update :formatted str "\n")
+
+        (not (get-in rules-config [:eof-newline :trailing-blanks?]))
+        (update :formatted str/replace-first #"\n*\z" "\n")))))
```

Comment:
A nested `cond->` works okay here. This might be able to be simplified a little bit using `as->` since the regex also fixes the no-trailing-newlines case:
```suggestion
      (as-> result
        (if (and (get-in rules-config [:eof-newline :trailing-blanks?])
                 (not (str/ends-with? (:formatted result) "\n")))
          (update result :formatted str "\n")
          (update result :formatted str/replace-first #"\n*\z" "\n"))))))
```
At this point it'd also make sense to me to factor this bit of logic out into a separate function which would just play nicely with the outer `cond->` though.